src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -1,30 +1,31 @@
 import { useAuth0 } from '@auth0/auth0-react';
 import { useRouter } from 'next/router';
-import React, { ReactElement } from 'react';
+import React, { FC, useContext, useMemo } from 'react';
 import { getAccountInfo } from '../../../utils/apiRequests/api';
-import { User } from '../types/user';
+import { User } from '@planet-sdk/common/build/types/user';
+import { SetState } from '../types/common';
 
-interface Props {}
+interface UserPropsContextInterface {
+  contextLoaded: boolean;
+  setContextLoaded: SetState<boolean>;
+  token: string | null;
+  setToken: SetState<string | null>;
+  user: boolean | User | null;
+  setUser: SetState<boolean | User | null>;
+  userLang: string;
+  setUserLang: SetState<string>;
+  isImpersonationModeOn: boolean;
+  setIsImpersonationModeOn: SetState<boolean>;
+  loginWithRedirect: (value: {}) => {};
+  logoutUser: (value: string) => void;
+  loadUser: () => {};
+}
```

Comment:
The context interface should match the values made available via that context. There are some issues here.

1. setters that are not being used in the app do not need to be returned back, and should be left out in the interface and within the `value` returned. This includes `setContextLoaded`, `setUserLang`. Review other setters as well. 
2. `auth0User`, `auth0Error` are missing in the interface, but are present in the `value` returned.
3. Function type for `loginWithRedirect` can match the type returned by `useAuth0()`
4. While defining function types, try to be exact. For example, logoutUser defined in the UserPropsProvider scope is slightly different in type to that defined in the interface
5. `loadUser` could be defined as `() => Promise<void>`. Here, you've defined it as an empty function.

Reply:
Made minor updates

[4849d57](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1750/commits/4849d57a67d18b21980901c5a834b1b33620c2a5)

[2761a9e](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1750/commits/2761a9e76f001f031a45f753ee893933224ce7c7)